### DISCUSSION TITLE
How can I use env variables in vercel.json file (if possible)?

### DISCUSSION THREAD

**Discussion by probablyArth** [vercel/vercel#9611] [Help]
Is something like this possible?

```json
{
  "crons": [
    {
      "path": "/api/cron/dailyQuote/${SECRET_KEY}",
      "schedule": "${NEXT_PUBLIC_JOB_TIME}"
    }
  ]
}
```

**Comment by johnpolacek** [comment]
It appears there is no secure way for us to use vercel cron jobs on an open source project, as your `vercel.json` file needs be in your project repo and there is no apparent means to use a secret. Am I missing something?

**Comment by TechEclipse** [comment]
At least for the authentication key you can use `CRON_SECRET` as production environment variable. This is yet not documented but mentioned in this discussion [here](https://github.com/orgs/vercel/discussions/1734#discussioncomment-6146073). For other environment variables like defining the value of `schedule` I don't see any solutions.
